### PR TITLE
get_requested_site: configurable backend to get the site for an API call

### PR DESCRIPTION
**Pull request by OmarIthawi**
### Description

Dashboard needs to be able to access Figures API with privileged access by sending site's UUID parameter.

This concept is Tahoe-specific and shouldn't make it into Figures (thanks John for advocating for that). Therefor this PR adds a plugin to customize the logic we use to `get_current_site()` 

I've added a Figures plugin repo to do that: 

 - https://github.com/appsembler/tahoe-figures-plugins/pull/1

### Using the backend

 - Install a package which has the custom backend e.g. `pip install tahoe-figures-plugins` 
 - In the `server-vars.yml` configure Figures like below:

   ```yaml
   FIGURES:
     REQUESTED_SITE_BACKEND: 'tahoe_figures_plugins.sites:get_current_site_or_by_uuid'
   ```

### Backward compatibility

 - By default this PR changes nothing for both community and Tahoe installations
 - The tests works as-is for all Figures views
 - The plugin only used when configured

### Benchmark

I tested the `import_from_path` function and the load takes 0.0069ms per call if the backend is already imported in the Python module system on a staging server.

Loading the backend for the first time (e.g. a new Python gunicorn worker) takes 0.5ms.

This seems like an acceptable delay to add on top of the each request.

<details><summary>Expand to see the "timeit" code used</summary>

**Test module reload time with cache**

```py
import timeit

load_with_cache_setup = """
from django.conf import settings
from figures.sites import import_from_path
import importlib
"""

load_with_cache = """
sites_backend_path = settings.ENV_TOKENS['FIGURES'].get('SITES_BACKEND')
sites_backend = import_from_path(sites_backend_path)
"""

print('load_with_cache', timeit.timeit(
    setup=load_with_cache_setup,
    stmt=load_with_cache,
    number=100000,
))

# load_with_cache 0.690 sec for 100k loads
```

**Test module reload time without cache**

```py
import timeit

load_without_cache_setup = """
from django.conf import settings
from figures.sites import import_from_path
import importlib
import openedx.core.djangoapps.appsembler.sites.utils
"""

load_without_cache = """
sites_backend_path = settings.ENV_TOKENS['FIGURES'].get('SITES_BACKEND')
importlib.reload(openedx.core.djangoapps.appsembler.sites.utils)
sites_backend = import_from_path(sites_backend_path)
"""

print('load_without_cache', timeit.timeit(
    setup=load_without_cache_setup,
    stmt=load_without_cache,
    number=10000,
))

# load_without_cache 5.033 sec for 10k loads
```
 </details>

### CODE REVIEW

**Review by estherjsuh — APPROVED** [review]

File: figures/sites.py [hunk 1]
```diff
@@ -304,6 +305,24 @@ def _get_all_sites():
     return Site.objects.all()
 
 
+def get_requested_site(request):
+    """
+    From a request return the corresponding site.
+
+    This functions makes use of the `REQUESTED_SITE_BACKEND` setting if configured, otherwise
+    it defaults to Django's get_current_site().
+
```
Comment (johnbaldwin) [inline figures/sites.py:314]: @OmarIthawi Please add  "Using the backend" instructions either here, in this module docstring or in a readme doc

**Review by johnbaldwin — APPROVED** [review]
Just one comment to include this new functionality configuration instructions in the code or a readme doc. Thanks!

File: figures/sites.py (hunk 1 above)
Comment (OmarIthawi) [reply figures/sites.py:314]: Thanks John! I'll add the instructions to the README doc.

Comment (OmarIthawi) [reply figures/sites.py:314]: I've documented the feature in a new Configurations section. I'll merge after someone spot check the section.

It's not a complete one, but a start.

**Review by thraxil — APPROVED** [review]

**Pull request merged**